htlcswitch/switch.go

```diff
@@ -872,43 +851,25 @@ func (s *Switch) handleLocalResponse(pkt *htlcPacket) {
 	// has been restarted since sending the payment.
 	payment := s.findPayment(pkt.incomingHTLCID)
```

Comment:
Suppose `lnd` crashes right at this point. What will happen after restart? `Router` knows that the payment was still in flight. I asks for the payment result and gets back payment id unknown. Then it starts a new attempt. But because the circuit is already torn down, switch will start a double payment?

Reply:
Question answered. Is a gap in this PR, but will be dealt with in the follow up switch pr.